src/protocol/libp2p/ping/mod.rs

```diff
@@ -141,14 +149,22 @@ impl Ping {
         tracing::trace!(target: LOG_TARGET, ?peer, "handle inbound substream");
 
         self.pending_inbound.push(Box::pin(async move {
-            let payload = substream
-                .next()
-                .await
-                .ok_or(Error::SubstreamError(SubstreamError::ReadFailure(None)))??;
-            substream.send_framed(payload.freeze()).await?;
-            let _ = substream.next().await.map(|_| ());
-
-            Ok(())
+            let future = async move {
+                let payload = substream
+                    .next()
+                    .await
+                    .ok_or(Error::SubstreamError(SubstreamError::ReadFailure(None)))??;
+                substream.send_framed(payload.freeze()).await?;
+                let _ = substream.next().await.map(|_| ());
```

Comment:
What are we reading here?

Reply:
Nothing, it's just some old code back when I was debugging the ping protocol and [it didn't work without it](https://i.makeagif.com/media/11-28-2016/XaxcSr.mp4). Probably can be removed.